doc/services/portability/posix/option_groups/index.rst

```diff
@@ -222,7 +222,7 @@ POSIX_DEVICE_IO
     feof(),
     ferror(),
     fflush(),
-    fgetc(),
+    fgetc(),yes (will fail with ``ENOSYS``:ref:`†<posix_undefined_behaviour>`)
```

Comment:
Technically, this shouldn't fail, as long as there is an open `FILE*`. But what happens when we do an `fopen()`? What path would we even pass to `fopen()` to get a `FILE*`? Is `stdin` already available? It should probably be (stdin, stdout, and stderr should be opened before `main()` in ISO C, but that is part of the c runtime).

That's just ISO C.

I think we are putting the cart before the horse so to speak.

The issue is that I don't believe we support file based streams (yet) in the libc / os level. We only support reading / writing to standard targets by integer file descriptor.

We might need to modify fdtable.c to allow us to use FILE based streams as well.

This seems to be a requirement for POSIX_DEVICE_IO in any case.